addons/mail/models/mail_message.py

```diff
@@ -941,92 +941,68 @@ def _notify_moderators(self):
     # MESSAGE READ / FETCH / FAILURE API
     # ------------------------------------------------------
 
-    @api.model
-    def _message_read_dict_postprocess(self, messages, message_tree):
-        """ Post-processing on values given by message_read. This method will
-            handle partners in batch to avoid doing numerous queries.
+    def _message_read_dict_postprocess(self, vals_list):
```

Comment:
Why don't you pass the list of fields, and read the data on `self`?
You can then format the fields by using `field.convert_to_read()` or any formatting you like.

Reply:
It would be similar to calling `_read_format` *after* `_message_read_dict_postprocess` instead of *before* it.

Reply:
As discussed, ok to move `_read_format` inside the method, but it still has to be called at the start and not at the end since it reads fields specifically and then we apply changes on top of that (of keys that are fields but also on new keys that are not fields).